log4j-nosql/src/main/java/org/apache/logging/log4j/nosql/appender/lucene/LuceneAppender.java

```diff
@@ -0,0 +1,301 @@
+		if (this.expiryTime != null){
+			registerClearTimer();
+		}
+	}
+
+	/**
+	 * IndexWriter initialization.
+	 */
+	private IndexWriter initIndexWriter() {
+		if (null == writerMap.get(target)) {
+			try {
+				FSDirectory fsDir = FSDirectory.open(Paths.get(this.target));
+				IndexWriterConfig writerConfig = new IndexWriterConfig(new LuceneAnalyzer());
+				writerMap.putIfAbsent(target, new IndexWriter(fsDir, writerConfig));
+			} catch (IOException e) {
+				LOGGER.error("IndexWriter initialization failed!" + e.getMessage(), e);
+			}
+		}
+		return writerMap.get(target);
+	}
+
+	/**
+	 * Register IndexWriter commit timertask.
+	 */
+	private void registerCommitTimer() {
+		scheduledExecutor.scheduleAtFixedRate(new Runnable() {
+			@Override
+			public void run() {
+				IndexWriter indexWriter = initIndexWriter();
+				if (null!=indexWriter && indexWriter.numRamDocs()>0) {
+					try {
+						indexWriter.commit();
+					} catch (IOException e) {
+						LOGGER.error("IndexWriter commit failed!"+e.getMessage(),e);
+					}
+				}
+			}
+		}, 1, 1, TimeUnit.MINUTES);
+	}
+	
+	/**
+	 * Register IndexWriter clean timertask.
+	 * Delete the index before {@link LuceneAppender#expiryTime} second every day at 0.
+	 * 
+	 * @see LuceneAppender#expiryTime
+	 */
+	private void registerClearTimer() {
+		Calendar calendar = Calendar.getInstance();
+		long curMillis = calendar.getTimeInMillis();
+		calendar.add(Calendar.DAY_OF_MONTH, 1);
+		calendar.set(Calendar.HOUR_OF_DAY, 0);
+		calendar.set(Calendar.MINUTE, 0);
+		calendar.set(Calendar.SECOND, 0);
+		calendar.set(Calendar.MILLISECOND, 0);
+		long difMinutes = (calendar.getTimeInMillis() - curMillis) / (1000 * 60);
+
+		scheduledExecutor.scheduleAtFixedRate(new Runnable() {
+			@Override
+			public void run() {
+				LOGGER.info("delete index start!",target,expiryTime);
+				IndexWriter indexWriter = initIndexWriter();
+				if (null != indexWriter) {
+					Long start = 0L;
+					Long end = System.currentTimeMillis() - expiryTime * 1000;
+					NumericRangeQuery<Long> rangeQuery = NumericRangeQuery.newLongRange("timestamp", start, end, true, true);
+					try {
+						indexWriter.deleteDocuments(rangeQuery);
+						indexWriter.commit();
+						LOGGER.info("delete index end! ",target,expiryTime);
+					}catch (IOException e) {
+						LOGGER.error("delete index failed! "+e.getMessage(),e);
+					}
+				}
+			}
+		}, difMinutes, 1440, TimeUnit.MINUTES);
+	}
+
+	/**
+	 * create lucene index.
+	 * 
+	 * @param event
+	 */
+	@Override
+	public void append(LogEvent event) {
+		if (null != indexFields && indexFields.length > 0) {
+			IndexWriter indexWriter = initIndexWriter();
+			if(null != indexWriter){
+				Document doc = new Document();
+				doc.add(new LongField("timestamp", event.getTimeMillis(), Field.Store.YES));
+				try {
+					for (LuceneIndexField field : indexFields) {
+						String value = field.getLayout().toSerializable(event);
+						if (Strings.isEmpty(value) || value.matches("[$]\\{.+\\}")) {

```

Comment:
Generally like to precompile regular expressions into Pattern objects. That's actually one case where a static field makes sense come to think of it. ;)